agent/worker.go

```diff
@@ -42,17 +50,19 @@ type worker struct {
 	db        *bolt.DB
 	executor  exec.Executor
 	listeners map[*statusReporterKey]struct{}
+	secrets   *Secrets
```

Comment:
If we added the Get/Add/Remove apis to secrets, seems like they should have pointer receivers because they're mutating internal state.  In that case, should this be a pointer in order to call those APIs?